### ISSUE TITLE
Load multiple ifc into a scene

### ISSUE THREAD

**Issue by MarkusSteinbrecher** [bldrs-ai/Share#248]
As a user I want to be able to upload multiple ifc models into a scene and be able to toggle each (e.g. architecture model and HVAC model).

**Comment by OlegMoshkovich** [comment]
@pablo-mayrgundter - does the work you did with the cars, address this issue?

**Comment by pablo-mayrgundter** [comment]
No.  I think we want to use the IFCjs loader to handle multiple models.  Let's leave this open to track that.

**Comment by MarkusSteinbrecher** [comment]
Apparently this is already possible with ifc.js: https://stackoverflow.com/questions/68222208/how-to-load-multiple-files-in-ifc-js

**Comment by MarkusSteinbrecher** [comment]
Updated the description. I propose to start with the easiest version in the UI and just add a "add model" to the "Open IFC" screen. What do you think?

**Comment by pablo-mayrgundter** [comment]
I think we have two related concepts here: Levels of a building and Layers of a building.  "Show me MEP & Architecture for level 4 of the building" includes both of these.

I propose 2 tool icons (on right side), one for each:
  - Section tool has the usual stack of squares, and allows toggling floors on and off.  
  - Layer tool is next to it, different icon, and allows toggling e.g. MEP & architecture on and off.
    - Layer tool also includes a "+" symbol, which triggers a file upload of an IFC that creates a new layer, that can also be toggled.

Separately, loading files in the top-level search dialog area is always "start a new project by removing all current content", or perhaps there's a checkbox to leave current content.  However, that raises the problem of whether the new content is in its own layer, and if so why not use the layer tool approach.

Talked with Oleg on this.. think we need some mocks.

**Comment by MarkusSteinbrecher** [comment]
Not sure I agree to the "layer tool" ... this is about loading another ifc model into a scene. The "layer tool" in my view would be one ifc model but I mark MEP as a layer inside that. Maybe this is just naming but if I want to upload another ifc into a scene I am not searching for "layer tool".
Regarding the mock ups: I am sure we can come up with a nice ui over time ... for now adding a "load additional model" on the open screen would work I think.

**Comment by pablo-mayrgundter** [comment]
Ok, as discussed, this is really just loading top-level models, i.e. with separate trees, no particular merging of layers (how do we know what's a layer?).  Good good

**Comment by Adrian62D** [comment]
Fits to product strategy: yes
Estimation: needs R&D work for Layering, potentially Q1

**Comment by Adrian62D** [comment]
@pablo-mayrgundter do you see an issue if we start with this one already even though the core refactoring on your end is in progress?

**Comment by pablo-mayrgundter** [comment]
Really depends how complex this will be.  If it's pretty simple, then it's ok as we're not committing too much work before the new datamodel is active.  So I'd suggest a day or two looking at it to see if there's a quick approach.

**Comment by Adrian62D** [comment]
We analyzed the requirements and updated the issue description accordingly. We think it's doable with an OK-ish amount of effort and will tackle this one.

**Comment by pablo-mayrgundter** [comment]
Good to hear.

IFCjs has support for this, and we'll be following that API for the new datamodel.
